mava/evaluator.py

```diff
@@ -104,41 +104,68 @@ def not_done(carry: Tuple) -> bool:
 
         return eval_metrics
 
-    def evaluator_fn(trained_params: FrozenDict, key: chex.PRNGKey) -> ExperimentOutput[EvalState]:
-        """Evaluator function."""
+    def parallel_evaluator(
+        carry: Tuple[FrozenDict, chex.PRNGKey], _: Any
+    ) -> Tuple[Tuple[FrozenDict, chex.PRNGKey], ExperimentOutput[EvalState]]:
+        """Evaluates the agent on multiple environments in parallel."""
+        trained_params, key = carry
 
-        # Initialise environment states and timesteps.
         n_devices = len(jax.devices())
+        episodes_per_device = config.arch.num_eval_episodes * eval_multiplier // n_devices
```

Comment:
Sadly jax doesn't allow us to carry these fixed values 😢

Reply:
I am quite sure there is a better way to implement this, but we can keep it as it is at the moment and create an issue calling for cleaning and readibility of the evaluator